src/reducers/workspaces.ts

```diff
@@ -167,25 +251,24 @@ export const reducer: Reducer<IWorkspaceManagerState> = (
     case SEND_REPL_INPUT_TO_OUTPUT:
       // CodeOutput properties exist in parallel with workspaceLocation
       newOutput = state[location].output.concat(action.payload as CodeOutput)
-      return {
-        ...state,
-        [location]: {
-          ...state[location],
-          output: newOutput
-        }
+      let newReplHistoryRecords: string[]
+      if (action.payload.value !== '') {
+        newReplHistoryRecords = [action.payload.value].concat(state[location].replHistory.records)
+      } else {
+        newReplHistoryRecords = state[location].replHistory.records
       }
-    case EVAL_EDITOR:
```

Comment:
I should have added some documentation for this, my bad. It is here to force a re-render in the workspace component, showing the "Stop" button. Can it be added back? The Stop button does not show up when running e.g `const fib = (x) => x <= 2? 1 : fib(x-1) + fib(x-2); fib(20);`